Source/Library/GameModel.cpp

```diff
@@ -2,16 +2,54 @@
 
 using namespace TicTacToe;
 
-GameModel::GameModel() {}
+GameModel::GameModel() : 
+  m_board(),
+  m_turn(Board::Token::X),
+  m_state(State::INVALID),
+  m_moves(0) {
+}
 
 const Board& GameModel::get_board() const {
-  throw std::runtime_error("Not implemented.");
+  return m_board;
 }
 
 Board::Token GameModel::get_turn() const {
-  throw std::runtime_error("Not implemented.");
+  return m_turn;
 }
 
 GameModel::State GameModel::play_move(int x, int y) {
-  throw std::runtime_error("Not implemented.");
+  if(m_state != State::END) {
+    if(m_board.get_piece(x, y) == Board::Token::NONE) {
+      m_board.set_piece(x, y, m_turn);
+      ++m_moves;
+      auto turn = [&] {
+        if(m_turn == Board::Token::X) {
+          m_turn = Board::Token::O;
+        } else if(m_turn == Board::Token::O) {
+          m_turn = Board::Token::X;
+        }
+      };
+      auto continue_play = [&] {
+        m_state = State::CONTINUE;
+        turn();
+      };
+      if(m_moves >= Board::ROW_COUNT) {
```

Comment:
Identify ways to eliminate a whole chunk of redundant checks and if statements, for example:

    if m_moves >= ROW_COUNT and evaluate_winner(m_board) != Board::Token::NONE or
        m_moves == ROW_COUNT * COLUMN_COUNT:
      m_state = State::END
    else:
      ++m_moves
      if m_turn == X:
        m_turn = O
      else:
        m_turn = X